xrdsst/controllers/token.py

```diff
@@ -54,11 +53,12 @@ def init_keys(self):
         urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
         self.token_add_keys_with_csrs(self.load_config())
 
-    def token_list(self, configuration):
-        self.init_logging(configuration)
-        for security_server in configuration["security-server"]:
-            configuration = self.initialize_basic_config_values(security_server)
+    def token_list(self, config):
+        self.init_logging(config)
+        for security_server in config["security-server"]:
+            configuration = self.initialize_basic_config_values(security_server, config)
             self.remote_token_list(configuration)
+            self.revoke_api_key(security_server, config)
```

Comment:
Please refactor scattered ``revoke_api_key`` calls to aspect. One possible approach would be to register created API keys and then perform revocation on app shutdown. Cement hooks (https://docs.builtoncement.com/core-foundation/hooks) would be one way to realized the aspect, most fitting probably being ``pre_close`` or ``post_close``.